geopolars/src/geoseries.rs

```diff
@@ -640,6 +647,35 @@ impl GeoSeries for Series {
         }
     }
 
+    fn distance(&self, other: &Series) -> Result<Series> {
+        use geo::algorithm::EuclideanDistance;
+
+        let mut output_array = MutablePrimitiveArray::<f64>::with_capacity(self.len());
+
+        for (g1, g2) in iter_geom(self).zip(iter_geom(other)) {
+            let distance = match (g1, g2) {
+                (Geometry::Point(p1), Geometry::Point(p2)) => Some(p1.euclidean_distance(&p2)),
+                (Geometry::Point(p1), Geometry::LineString(p2))=>Some(p1.euclidean_distance(&p2)),
+                (Geometry::Point(p1), Geometry::MultiLineString(p2))=>Some(p1.euclidean_distance(&p2)),
+                (Geometry::Point(p1), Geometry::MultiPoint(p2))=>Some(p1.euclidean_distance(&p2)),
+                (Geometry::Point(p1), Geometry::MultiPolygon(p2))=>Some(p1.euclidean_distance(&p2)),
+                (Geometry::Point(p1), Geometry::Polygon(p2))=>Some(p1.euclidean_distance(&p2)),
+
+                (Geometry::LineString(p1), Geometry::Point(p2))=>Some(p2.euclidean_distance(&p1)),
+                (Geometry::MultiLineString(p1), Geometry::Point(p2))=>Some(p2.euclidean_distance(&p1)),
+                (Geometry::MultiPoint(p1), Geometry::Point(p2))=>Some(p2.euclidean_distance(&p1)),
+                (Geometry::MultiPolygon(p1), Geometry::Point(p2))=>Some(p2.euclidean_distance(&p1)),
+                (Geometry::Polygon(p1), Geometry::Point(p2))=>Some(p2.euclidean_distance(&p1)),
+                _ => None,
```

Comment:
What happens when neither of the two geometries are of type `Point`? There are `impl`s for it looks like every combination of geometries: https://docs.rs/geo/latest/geo/algorithm/euclidean_distance/trait.EuclideanDistance.html#impl-EuclideanDistance%3CT%2C%20LineString%3CT%3E%3E-for-Polygon%3CT%3E

I don't know the best way to handle these functions where we want to support the cartesian product between each geometry type and each other geometry type. I hope we don't have to manually list out every combination of geometry types every time, like we do here:
https://github.com/geopolars/geopolars/blob/6c28420581866fb38745dd6044c615a3f4b93c2c/geopolars/src/spatial_index.rs#L86-L132

Reply:
for some reason i though that I had them all, didnt see impl for other classes.

I don't know the better way. I really wanted to do it differently because for sure it doesn't scale. More so that It is symetric relation so if there is no implementation X,Y=d i need to check if there is Y,X=d. And i think it happens more often.

And all of those types use the same stuff e.g `poly_lhs.intersects(poly_rhs)`. I dont know how in rust you would work with that. I guess some macro would be nice

Reply:
@kylebarron  I can create such macro in another PR if you specify it how it should work because that two cases differ a bit. For now i added missing traits. But i dont like it that had to copy paste now the same stuff.

Reply:
I don't have much experience with Rust macros. Definitely not necessary for this PR. Eventually I'd like to figure out a better solution